nima/http/encoding/encoding/src/main/java/io/helidon/nima/http/encoding/ContentEncodingContext.java

```diff
@@ -88,4 +96,97 @@ static ContentEncodingContext create() {
      * @return content encoder to use
      */
     ContentEncoder encoder(Headers headers);
+
+    /**
+     * Builder to set up this encoding support content.
+     *
+     * @return a new builder
+     */
+    static Builder builder() {
+        return new Builder();
+    }
+
+    /**
+     * Fluent API builder for {@link ContentEncodingContext}.
+     */
+    class Builder implements io.helidon.common.Builder<Builder, ContentEncodingContext> {
+
+        private static final String IDENTITY_ENCODING = "identity";
+
+        private final HelidonServiceLoader.Builder<ContentEncodingProvider> encodingProviders
+                = HelidonServiceLoader.builder(ServiceLoader.load(ContentEncodingProvider.class));
+
+        /**
+         * Update this builder from configuration.
+         * <p>
+         * Configuration:<ul>
+         *     <li><b>disable: true</b>  - to disable content encoding support</li>
+         * </ul>
+         *
+         * @param config configuration to use
+         * @return updated builder instance
+         */
+        public Builder config(Config config) {
+            config.get("discover-services").asBoolean().ifPresent(this::discoverServices);
+            return this;
+        }
+
+        /**
+         * Disable content encoding support.
```

Comment:
This is not a correct javadoc. It disables discovery from service loader only.

Reply:
fixed